### PR TITLE
[AzureMonitorExporter] capture error message from ingestion in logging

### PR DESCRIPTION
**Pull request by TimothyMothra**
Towards #34581

### Changes
- added helper class to deserialize and parse the error messages from ingestion response
- refactor eventsource method used when a Transmission fails.
  - includes error message from ingestion response

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/IngestionResponseHelper.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Monitor.OpenTelemetry.Exporter.Internals.Diagnostics;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    internal static class IngestionResponseHelper
+    {
+        public static string?[] GetErrorsFromResponse(Response? response)
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/IngestionResponseHelper.cs:15]: Could you please validate the logic from the below code and confirm if its on the same lines. https://github.com/microsoft/ApplicationInsights-dotnet/blob/248800626c1c31a2b4100f64a884257833b8c77f/BASE/src/ServerTelemetryChannel/Implementation/TransmissionPolicy/ErrorHandlingTransmissionPolicy.cs#L40C89-L40C89

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs [hunk 2]
```diff
@@ -207,15 +207,21 @@ internal static ExportResult HandleFailures(HttpMessage httpMessage, PersistentB
                 }
             }
 
-            AzureMonitorExporterEventSource.Log.TransmissionFailed(fromStorage: false, statusCode: statusCode, connectionVars: connectionVars, requestEndpoint: httpMessage.Request.Uri.Host, willRetry: (result == ExportResult.Success));
+            AzureMonitorExporterEventSource.Log.TransmissionFailed(
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs:210]: What is the cost of logging a message when a listener is enabled and the batch fail all the times?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs:210]: Is it possible to asynchronously log this message?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs:210]: > what is the cost

If this is happening repeatedly, the Exporter should backoff.

> is it possible to asynchronously log

I don't know, i'll look into this :)

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs:210]: > If this is happening repeatedly, the Exporter should backoff.

May not be true, if its a partial failure.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/IngestionResponseHelper.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/IngestionResponseHelper.cs:15]: It's very similar, almost identical datamodels but using a different deserializer.
AI SDK only does the deserialization when logging is Verbose. I think this is a good idea and would mitigate our perf concerns.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs (hunk 2 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs:210]: mitigating this by only deserializing errors when LogLevel is set to Verbose

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/IngestionResponseHelper.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            [JsonPropertyName("itemsReceived")]
+            public int ItemsReceived { get; set; }
+
+            [JsonPropertyName("itemsAccepted")]
+            public int ItemsAccepted { get; set; }
+
+            [JsonPropertyName("errors")]
+            public List<ErrorObject>? Errors { get; set; }
+        }
+        private class ErrorObject
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/IngestionResponseHelper.cs:53]: nit: consider a line return between 2 classes.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Comment by rajkumar-rangaraj** [comment]
Collect data using logman / perfview to validate if the verbose or critical level log is honored.

**Comment by TimothyMothra** [comment]
> Collect data using logman / perfview to validate if the verbose or critical level log is honored.

I tested this with both the OTel Self-Diagnostics & our DebugTelemetryWriter.

**Pull request merged**